### PR TITLE
Add -d option for finding duplicates

### PR DESCRIPTION
**Pull request by viraptor**
Cargo has problems with finding optimal set of packages to download. Sometimes
this will result in multiple versions of the same dependency being applied in
the same project and lead to cryptic errors. (rust-lang/cargo#2064)

-d option will find all packages using the same name. If they exist in more than
one version, they'll be printed out in the "inverted" mode, showing where did
they get imported.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -165,11 +169,52 @@ fn real_main(flags: Flags, config: &Config) -> CliResult<Option<()>> {
         Charset::Utf8 => &UTF8_SYMBOLS,
     };
 
-    print_tree(root, kind, &graph, direction, symbols, flag_all);
+    if flag_duplicates {
+        let dups = find_duplicates(&graph);
+        if dups.len() == 0 {
+            println!("No package name is imported more than once");
```
Comment (sfackler) [inline src/main.rs]: I would lean towards leaving this out personally - things like `cargo build` produce no output if there's nothing to do.

**Comment by sfackler** [comment]
This seems cool! One little comment but lgtm otherwise.

**Comment by viraptor** [comment]
Fixed now.

**Comment by sfackler** [comment]
Thanks!

**Pull request merged**